src/Features/Core/Portable/EmbeddedLanguages/DateTime/DateTimeEmbeddedCompletionProvider.cs

```diff
@@ -0,0 +1,219 @@
+            context.AddCustom("fff", "1,000ths of a second", "blah blah");
+            context.AddCustom("ffff", "10,000ths of a second", "blah blah");
+            context.AddCustom("fffff", "100,000ths of a second", "blah blah");
+            context.AddCustom("ffffff", "1,000,000ths of a second", "blah blah");
+            context.AddCustom("fffffff", "10,000,000ths of a second", "blah blah");
+
+            context.AddCustom("F", "10ths of a second (non-zero)", "blah blah");
+            context.AddCustom("FF", "100ths of a second (non-zero)", "blah blah");
+            context.AddCustom("FFF", "1,000th of a second (non-zero)", "blah blah");
+            context.AddCustom("FFFF", "10,000ths of a second (non-zero)", "blah blah");
+            context.AddCustom("FFFFF", "100,000ths of a second (non-zero)", "blah blah");
+            context.AddCustom("FFFFFF", "1,000,000ths of a second (non-zero)", "blah blah");
+            context.AddCustom("FFFFFFF", "10,000,000ths of a second (non-zero)", "blah blah");
+
+            context.AddCustom("gg", "period/era", "blah blah");
+
+            context.AddCustom("h", "12 hour clock (1-2 digits)", "blah blah");
+            context.AddCustom("hh", "12 hour clock (2-digits)", "blah blah");
+
+            context.AddCustom("H", "24 hour clock (1-digit)", "blah blah");
+            context.AddCustom("HH", "24 hour clock (2-digits)", "blah blah");
+
+            context.AddCustom("K", "time zone", "blah blah");
+
+            context.AddCustom("m", "minute (1-2 digits)", "blah blah");
+            context.AddCustom("mm", "minute (2 digits)", "blah blah");
+
+            context.AddCustom("M", "month (1-2 digits)", "blah blah");
+            context.AddCustom("MM", "month (2 digits)", "blah blah");
+            context.AddCustom("MMM", "month (abbreviated)", "blah blah");
+            context.AddCustom("MMMM", "month (full)", "blah blah");
+
+            context.AddCustom("s", "second (1-2 digits)", "blah blah");
+            context.AddCustom("ss", "second (2 digits)", "blah blah");
+
+            context.AddCustom("t", "AM/PM (abbreviated)", "blah blah");
+            context.AddCustom("tt", "AM/PM (full)", "blah blah");
+
+            context.AddCustom("y", "year (1-2 digits)", "blah blah");
+            context.AddCustom("yy", "year (2 digits)", "blah blah");
+            context.AddCustom("yyy", "year (3-4 digits)", "blah blah");
+            context.AddCustom("yyyy", "year (4 digits)", "blah blah");
+            context.AddCustom("yyyyy", "year (5 digits)", "blah blah");
+
+            context.AddCustom("z", "utc hour offset (1-2 digits)", "blah blah");
+            context.AddCustom("zz", "utc hour offset (2 digits)", "blah blah");
+            context.AddCustom("zzz", "utc hour and minute offset", "blah blah");
+
+            context.AddCustom(":", "time separator", "blah blah");
+            context.AddCustom("/", "date separator", "blah blah");
```

Comment:
how likely is it that more get added?  I'm going to assume there's not an authoritative source that we can get this from

Reply:
unlikely.